webapp/src/ts/services/route-snapshot.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Router } from '@angular/router';
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RouteSnapshotService {
+  constructor(
+    private router:Router,
+  ) {
+  }
+
+  get() {
```

Comment:
Oh nice! I think I added a comment before about having a service for this kind of things, what do you think if we add a function to return the must updated route and params, so we don't worry about thinking "do I need to subscribe to changes or snapshot will give me what's needed?" We have several places where we can use this

Reply:
Yea, I can refactor those places. In some cases we still need to subscribe, where we need to react to changes as they happen.

Reply:
On second though, I looked at the code again. 
We use the `this.routeSnapshot` in the view, so it needs to be responsive to changes, in this case. There isn't some trigger that will make us be able to call this new service and update the local route.